### ISSUE TITLE
Rename group tab if group renamed

### ISSUE THREAD

**Issue by chrisgoringe** [chrisgoringe/cg-controller#363]

**Comment by LukeG89** [comment]
Just tested and it's working with no problems!

**Comment by JorgeR81** [comment]
For me, sometimes it fails, in more complex workflows.

Sometimes it works, sometimes it doesn't. In the same group.

**Comment by LukeG89** [comment]
The only problem that I noticed is when updating the new title. After you edited it (I do it by double-clicking the group header), the tab will only update if you click on canvas or switch tabs. Otherwise, it doesn't update.


https://github.com/user-attachments/assets/469b1970-c7a0-477e-9030-74d123ffbd26

**Comment by LukeG89** [comment]
Found a bug when swiching workflows


https://github.com/user-attachments/assets/8f6fdd6d-6e53-4124-b5cd-dcd3d6cf3672

**Comment by chrisgoringe** [comment]
> The only problem that I noticed is when updating the new title. After you edited it (I do it by double-clicking the group header), the tab will only update if you click on canvas or switch tabs. Otherwise, it doesn't update.

Yes, this is the case for a few things. Problem is that Comfy doesn't provide a hook to know when things have happened, so we have to (a) hijack something (risk of incompatibility with changes or other nodes) (b) keep checking (performance impact) or (c) wait until the next time Comfy does say something has happened and look for changes.

I'm doing (c) where there isn't really a problem with a slight delay.

**Comment by chrisgoringe** [comment]
> Found a bug when switching workflows

Created #375 - this is part of a wider issue with the way Comfy handles extra information that I need to work out.

**Comment by chrisgoringe** [comment]
> For me, sometimes it fails, in more complex workflows.
> 
> Sometimes it works, sometimes it doesn't. In the same group.
> 
> After opening an workflow, it usually renames a group correctly, the first time. But the next time it fails.

Can you provide an example? I've renamed groups multiple times without a problem.

**Comment by chrisgoringe** [comment]
I can't reproduce any of the issues raised - can either of you work out anything more about the problems you are having?

**Comment by JorgeR81** [comment]
Yes, I just noticed the issue only happens if the Controller is hidden when I change the group name.

**Comment by chrisgoringe** [comment]
> Yes, I just noticed my issue only happens if the Controller is hidden when I change the group's name.

Ah, that makes sense. Raised as #376
